File: src/main/java/io/group02/fight4flight/customer/Customer.java
package io.group02.fight4flight.customer;

public class Customer {
    private String FName;
    private String LName;
    private String Email;
    private Long ID;
    // private Location Address;

    public Customer(String FName, String LName, String Email, Long ID) {
        this.FName = FName;
        this.LName = LName;
        this.Email = Email;
        this.ID = ID;
    }

    public String GetFName(){ return this.FName;}

    public String GetLName(){ return this.LName;}

    public String GetEmail(){ return this.Email;}
    
    public Long GetID() {
        return this.ID;}

    public void SetID(Long ID) {
        this.ID = ID;
    }
    // public GetAddress(){ return this.Address;}

    public void SetFName(String FName){ this.FName = FName;}

    public void SetLName(String LName){ this.LName = LName;}

    public void SetEmail(String Email){ this.Email = Email;}

    
}

File: src/main/resources/learning/src/Customer.js
import React, { useState } from 'react';
import TextField from '@mui/material/TextField';
import { Container, Paper, Button } from '@mui/material'

function Customer() {
    const [fname, setFName] = useState('')
    const [lname, setLName] = useState('')
    const [email, setEmail] = useState('')

    const handleClick = (e) => {
        e.preventDefault()
        const customer = { fname, lname, email }
        console.log(customer)
        console.log("POST SENDING")
        console.log(JSON.stringify(customer))
        fetch('http://localhost:8080/customer/add', {
            method: "POST",
            headers: { "Content-Type": "application/json" },
            body: JSON.stringify(customer)
        }
        ).then(() => {
            console.log("Customer was added")
        })
    }

    return (
        <Container>
            <Paper elevation={3}>
                <h1 style={{ color: "blue" }}>Add Customer</h1>
                <form>
                    <TextField label="First Name" variant="outlined" margin="normal" required fullWidth
                        value={fname}
                        onChange={(e) => setFName(e.target.value)}
                    />
                    <TextField label="Last Name" variant="outlined" margin="normal" required fullWidth
                        value={lname}
                        onChange={(e) => setLName(e.target.value)}
                    />
                    <TextField label="Email Address" variant="outlined" margin="normal" required type="email" fullWidth
                        value={email}
                        onChange={(e) => setEmail(e.target.value)}
                    />
                    <Button variant="contained" color="secondary" onClick={handleClick}>
                        Submit
                    </Button>
                </form>
            </Paper>
        </Container>
    );
}

export default Customer;
